service/history/queues/executable.go

```diff
@@ -401,7 +405,7 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 	}
 
 	// TODO: expand on the errors that should be considered terminal
-	if errors.As(err, new(TerminalTaskError)) {
+	if errors.As(err, new(TerminalTaskError)) || (e.dlqInternalErrors() && common.IsInternalError(err)) {
```

Comment:
Can we print out a separate log or metric for internal errors even if we are not dlq or dropping it. Then we can know it's safe to enable the feature flag. Like a shadow mode.